addons/mail/static/src/models/messaging_menu/messaging_menu.js

```diff
@@ -34,23 +34,38 @@ function factory(dependencies) {
         /**
          * Toggle whether the messaging menu is open or not.
          */
-        async toggleOpen() {
-            if (!this.isOpen) {
-                const inbox = this.env.messaging.inbox;
-                if (!inbox.mainCache.isLoaded && !inbox.mainCache.isLoading) {
-                    // populate some needaction messages on threads.
-                    // FIXME: await necessary due to bug in tests without it
-                    // see task-id 2275999
-                    await this.async(() => inbox.mainCache.loadMessages());
-                }
-            }
+        toggleOpen() {
             this.update({ isOpen: !this.isOpen });
         }
 
         //----------------------------------------------------------------------
         // Private
         //----------------------------------------------------------------------
 
+        /**
+         * @private
+         */
+        _computeInboxMessagesAutoloader() {
+            if (!this.isOpen) {
+                return;
+            }
+            const inbox = this.env.messaging.inbox;
+            if (!inbox) {
+                return;
+            }
+            if (!inbox.mainCache) {
+                return;
+            }
+            if (inbox.mainCache.isLoaded) {
+                return;
+            }
+            if (inbox.mainCache.isLoading) {
+                return;
+            }
```

Comment:
this is a bit hard to read, only 1 return would be better

Reply:
done